alicloud/data_source_alicloud_disks.go

```diff
@@ -0,0 +1,273 @@
+package alicloud
+
+import (
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/aliyun/alibaba-cloud-sdk-go/services/ecs"
+	"github.com/aliyun/alibaba-cloud-sdk-go/sdk/requests"
+	"regexp"
+	"fmt"
+	"log"
+)
+
+func dataSourceAlicloudDisks() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAlicloudDisksRead,
+
+		Schema: map[string]*schema.Schema{
+			"ids": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				ForceNew: true,
+				MinItems: 1,
+			},
+			"name_regex": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateNameRegex,
+				ForceNew:     true,
+			},
+			"type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validateDiskType,
```

Comment:
I think there is no need a new method `validateDiskType ` and can use `validateAllowedStringValue` instead.